### PR TITLE
Beta2

### PR DESCRIPTION
**Pull request by pimg**
merge beta2 into master so master branch is up to date again and can be used as a base for OVS and JIT development

### CODE REVIEW

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 1]
```diff
@@ -275,9 +275,12 @@ CREATE TABLE dcsa_im_v3_0.booking (
     vessel_id uuid NULL REFERENCES dcsa_im_v3_0.vessel(id),
     export_voyage_number varchar(50) NULL,
     place_of_issue varchar(100) NULL REFERENCES dcsa_im_v3_0.location(id),
-    pre_carriage_mode_of_transport_code varchar(3) NULL REFERENCES dcsa_im_v3_0.mode_of_transport(mode_of_transport_code)
+    pre_carriage_mode_of_transport_code varchar(3) NULL REFERENCES dcsa_im_v3_0.mode_of_transport(mode_of_transport_code),
+    valid_until timestamp with time zone NULL
 );
 
+CREATE UNIQUE INDEX unq_valid_until_booking_idx ON dcsa_im_v3_0.booking(carrier_booking_request_reference) WHERE valid_until IS NULL;
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:282]: We need to move this into 05 - it is not part of the "official" IM...

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 2]
```diff
@@ -287,9 +290,12 @@ CREATE TABLE dcsa_im_v3_0.shipment (
     carrier_id uuid NOT NULL REFERENCES dcsa_im_v3_0.carrier(id),
     carrier_booking_reference varchar(35) NOT NULL UNIQUE,
     terms_and_conditions text NULL,
-    confirmation_datetime timestamp with time zone NOT NULL
+    confirmation_datetime timestamp with time zone NOT NULL,
+    valid_until timestamp with time zone NULL
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:294]: We need to move this into 05 - it is not part of the "official" IM...

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 3]
```diff
@@ -351,9 +358,12 @@ CREATE TABLE dcsa_im_v3_0.shipping_instruction (
     displayed_name_for_port_of_load varchar(250) NULL,
     displayed_name_for_port_of_discharge varchar(250) NULL,
     displayed_name_for_place_of_delivery varchar(250) NULL,
-    amendment_to_transport_document varchar(20) NULL
+    amendment_to_transport_document uuid NULL,
+    valid_until timestamp with time zone NULL
 );
 
+CREATE UNIQUE INDEX unq_valid_until_si_idx ON dcsa_im_v3_0.shipping_instruction(shipping_instruction_reference) WHERE valid_until IS NULL;
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:365]: We need to move this into 05 - it is not part of the "official" IM...

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 4]
```diff
@@ -364,31 +374,34 @@ CREATE INDEX ON dcsa_im_v3_0.value_added_service_request (booking_id);
...
     received_for_shipment_date date NULL,
     number_of_originals integer NULL,
     issuer uuid NULL REFERENCES dcsa_im_v3_0.carrier(id),
-    shipping_instruction_id varchar(100) NOT NULL REFERENCES dcsa_im_v3_0.shipping_instruction (id),
+    shipping_instruction_id uuid NOT NULL REFERENCES dcsa_im_v3_0.shipping_instruction (id),
     declared_value_currency varchar(3) NULL,
     declared_value real NULL,
-    number_of_rider_pages integer NULL
+    number_of_rider_pages integer NULL,
+    valid_until timestamp with time zone NULL
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:389]: We need to move this into 05 - it is not part of the "official" IM...

File: datamodel/initdb.d/03_dcsa_im_v3_0.sql [hunk 5]
```diff
@@ -364,31 +374,34 @@ CREATE INDEX ON dcsa_im_v3_0.value_added_service_request (booking_id);
...
 );
 
+CREATE UNIQUE INDEX unq_valid_until_td_idx ON dcsa_im_v3_0.transport_document(transport_document_reference) WHERE valid_until IS NULL;
```
Comment (HenrikHL) [inline datamodel/initdb.d/03_dcsa_im_v3_0.sql:392]: We need to move this into 05 - it is not part of the "official" IM...

**Review by HenrikHL — APPROVED** [review]

**Review by mp-assco — APPROVED** [review]
looks good to me

**Pull request merged**